src/osx/cocoa/window.mm

```diff
@@ -2440,14 +2440,12 @@ When dragging the bottom part of the DND sample ("Drag text from here!")
     if ( wxpeer ) 
     {
         // since native rtti doesn't have to be enabled and wx' rtti is not aware of the mixin wxTextEntry, workaround is needed
-        wxTextCtrl *tc = wxDynamicCast( wxpeer , wxTextCtrl );
-        wxComboBox *cb = wxDynamicCast( wxpeer , wxComboBox );
-        if ( tc )
+        if ( wxTextCtrl *tc = wxDynamicCast( wxpeer , wxTextCtrl ) )
         {
             tc->MarkDirty();
             tc->SendTextUpdatedEventIfAllowed();
         }
-        else if ( cb )
+        else if ( wxComboBox *cb = wxDynamicCast( wxpeer , wxComboBox ) )
```

Comment:
I know it's close to your heart but does [No assignments in conditional expressions](http://wxwidgets.org/develop/coding-guidelines/#no_assignment_in_if) no longer apply or was an exception made here?

I'm not a fan but it does look slightly better here with regards to distance of declaration and usage of `cb`, without splitting up the `else if` and making it uglier in the process.

Reply:
I definitely don't consider this rule for initializing variables declared inside the `if`. The style guide needs to be updated, but this is (by far) not the only item which needs to be brought up to date there, so I'm a bit afraid of starting to change it...

Reply:
Thanks for the explanation. I didn't consider this distinction with initialization and that makes total sense.

As I don't have the burden of knowing where the style guide is wrong (I will try to use recent sources more for style decisions) I will look into changing this later.